spec/abstractops.html

```diff
@@ -1370,12 +1369,14 @@ <h1>
       <dd>It parses the argument as an ISO 8601 string and returns the calendar identifier, or *undefined* if there is none.</dd>
     </dl>
     <emu-alg>
-      1. Let _parseResult_ be ParseText(StringToCodePoints(_isoString_), |TemporalCalendarString|).
-      1. If _parseResult_ is a List of errors, throw a *RangeError* exception.
-      1. Let _id_ be the source text matched by the |CalendarName| Parse Node contained within _parseResult_, or an empty sequence of code points if not present.
-      1. If _id_ is empty, then
+      1. Let _calendar_ be ~empty~.
+      1. Set _calendar_ to ? ParseISODateTime(_isoString_).[[calendar]].
+      1. If _calendar_ is not *undefined*, return _calendar_.
+      1. Let _parseResult_ be ParseText(StringToCodePoints(_isoString_), |CalendarName|).
```

Comment:
We have to deal with the possibility that _parseResult_ is a List of errors (this happens when _isoString_ is not an ISO string and also not a valid |CalendarName|, e.g. `Temporal.Calendar.from('ninechars')`